### ISSUE TITLE
Make validation message optional in Input component

### ISSUE THREAD

**Issue by fbarl** [weaveworks/ui-components#247]
If we start using this component outside the forms context, we might expect a lot of inputs not doing validation.

If we want them to use them inline (like on the screenshot), the vertical spacing gets in the way.

![image](https://user-images.githubusercontent.com/1216874/41087181-44460daa-6a3c-11e8-8299-e6a5a2ded0c3.png)

A more systematic solution might be to support the `inline` mode, in which case the validation message could come after the input.

cc @jpellizzari @bia

**Comment by jpellizzari** [comment]
Agreed. Doing an absolute positioned validation message might make more
sense, so we don’t have to leave room for it on the page.

**Comment by jpellizzari** [comment]
Closed by https://github.com/weaveworks/ui-components/pull/277

Commit: 7ae284f9960fe60d0b2f85f8810b3f37b1b9868f
